mission_control/src/mission.cpp

```diff
@@ -36,359 +36,67 @@
 // Original version: Christopher Scianna Christopher.Scianna@us.QinetiQ.com
 
 #include "mission_control/mission.h"
-#include "mission_control/behavior.h"
+
 using mission_control::Mission;
-using mission_control::Behavior;
+using namespace mission_control;
 
-Mission::Mission() : missionState(MissionState::READY), m_current_behavior(NULL), elasped_time(0.0)
+Mission::Mission()
 {
-  current_abort_behavior_id = 0;
-  current_behavior_id = 0;
+  _missionFullPath = " Hola";
+  currentBehaviorID = 0;
+  
+  _missionFactory.registerNodeType<Waypoint1>("Waypoint1");
+  _missionFactory.registerNodeType<Attitude_Servo>("Attitude_Servo");
+  _missionFactory.registerNodeType<Depth_Heading>("Depth_Heading");
+
 }
 
 Mission::~Mission()
 {
-  if (m_behaviors.size() > 0)
-  {
-    for (int x = 0; x < m_behaviors.size(); x++)
-    {
-      delete m_behaviors[x];
-    }
-    m_behaviors.clear();
-  }
-
-  if (m_aborts.size() > 0)
-  {
-    for (int x = 0; x < m_aborts.size(); x++)
-    {
-      delete m_aborts[x];
-    }
-    m_aborts.clear();
-  }
 
-  current_abort_behavior_id = 0;
-  current_behavior_id = 0;
 }
 
-void Mission::Stop()
-{
-  if (GetState() == MissionState::EXECUTING)
-  {
-    SetState(MissionState::STOPPED);
-
-    // Wait for the mission processing thread to finish
-    ROS_INFO("Stopping mission processing thread");
-    m_threadProcMission->interrupt();  // wake up thread if sleeping
-    m_threadProcMission->join();
-
-    // Clean up all of the behavior publishers and messages
-    ROS_INFO("Stopping and cleaning up active behaviors");
-    Behavior* cur_behavior = getNextBehavior(true);
-    do
-    {
-      cur_behavior->stopBehavior();
-      cur_behavior = getNextBehavior();
-    }
-    while (cur_behavior != NULL);
-  }
-  else if (GetState() == MissionState::ABORTING)
-  {
-    SetState(MissionState::STOPPED);
-
-    ROS_INFO("Stopping abort mission thread");
-    m_threadAbortMission->interrupt();
-    m_threadAbortMission->join();
-
-    // Clean up all of the behavior publishers and messages
-    ROS_INFO("Stopping and cleaning up active behaviors");
-    Behavior* cur_behavior = getNextAbortBehavior(true);
-    do
-    {
-      cur_behavior->stopBehavior();
-      cur_behavior = getNextAbortBehavior();
-    }
-    while (cur_behavior != NULL);
+bool Mission::loadBehavior(const std::string &missionFullPath){
+//  TODO Check for erros and return -1 if were.
+  _missionFullPath = missionFullPath;
```

Comment:
@patapufate consider not storing the path until we know it's OK.

Reply:
Done